### ISSUE TITLE
Mount /a /a/b to the container and change the directory attribute in the container to implement container escape.

### ISSUE THREAD

**Issue by fw-ctf** [kubeedge/kubeedge#3532]
**What happened**:
This is a security breach.
**What you expected to happen**:
I want you to give me a new version to fix it or provide mitigations.
**How to reproduce it (as minimally and precisely as possible)**:
1. Create container A running as the root user and mount the /home/test directory of the host to the /home/tmp directory of container A.
2. In container A, run the mkdir command to create the childtest directory under the /home/tmp directory, which is equivalent to creating a childtest directory under the /home/test directory on the active node.
3. Create container B that runs as the root user, mount the /home/test directory of the host to the /home/tmp1 directory of container B, and mount the /home/test/childtest directory of the host to the /home/tmp2 directory of container B.
4. In container B, go to /home/tmp1, delete the childtest directory, and run the ln -s / ./childtest command to change the childtest directory to a soft link pointing to the root directory. In this case, the soft link points to the root directory in container B.
5. Restart container B and access the /home/tmp2 directory. It is found that the mounting path of container B is changed to the root directory of the host, implementing container escape.
**Anything else we need to know?**:
I hope to fix this vulnerability as soon as possible. Using kubeedge brings additional security risks.
**Environment**:
- Kubernetes version (use `kubectl version`):1.19.4
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):1.6.1(all)
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):ubuntu
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

**Comment by sdghchj** [comment]
How does kubelet solve this issue?

**Comment by liuwnzh** [comment]
> You mean kubeEdge have not implemented `MountPropagation` yet? `MountPropagation="HostToContainer"`,

No. You can mount the soft link of the host in the container to implement container escape.

**Comment by fw-ctf** [comment]
My product uses a trustlist to restrict the mounting directory of customers. For the first time, you can use soft links to bypass the trustlist check and fix this vulnerability.The customer bypassed our check again in the preceding scenario, so I think this issue should be resolved by the open source software itself.

**Comment by fisherxu** [comment]
In step5, restart containerB will start a new container?

**Comment by fisherxu** [comment]
I just test with `docker run`, I mount the `/home/test` to container `/home/tmp1`. Then in container under `/home/tmp1`, run `ln -s / ./childtest`. Then I found in container, the childtest points / of container. And in host, the childtest  points to / of host. So I think this is the design of container runtime? :)

**Comment by fisherxu** [comment]
Through the MountPropagation, the operation in pod/container will not reflict in host? @sdghchj

**Comment by fw-ctf** [comment]
Change the directory in container B to a soft link and restart the container by exhausting resources. When the container restarts, the directory pointed to by the soft link is mounted to container B to implement container escape.
This can be done with two containers, and in extreme cases one container can.
I think this is a CVE, not a bug, so the community should issue a CVE number.
